core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -260,9 +260,6 @@ class KafkaServer(
 
         /* generate brokerId */
         config._brokerId = getOrGenerateBrokerId(initialMetaPropsEnsemble)
-        // Currently, we are migrating from ZooKeeper to KRaft. If broker.id.generation.enable is set to true,
-        // we must ensure that the nodeId synchronizes with the broker.id to prevent the nodeId from being -1,
-        // which would result in a failure during the migration.
         config._nodeId = config.brokerId
```

Comment:
The `broker.id` must be defined in migration with this patch, so we don't need to update `_nodeId`. Could you please make `nodeId` be immutable again? Or just revert the change of #16540 :smile:

Reply:
Yes, fair point. Let's do that later since we're trying to get this specific change into 3.9. In trunk we'll be able to make both immutable.